woocommerce/woocommerce-android
# Hack week/stats widget - Step 1: Display current day v4 stats, if available, for the current site

This PR completes the first step of the Stats widget Project as defined in the master thread [here](https://github.com/woocommerce/woocommerce-android/issues/2041).

#### Changes
- Adds a new widget to the app that displays the current day stats, if v4 stats is available for the site.
- An error message is displayed in the widget if the user is not logged in to the app.
- An error message is displayed in the widget if the store does not have WC-Admin installed or if WooCommerce 4.0 is not installed.

#### Screenshot
##### Adding a new widget for site
<img width="300" src="https://user-images.githubusercontent.com/22608780/76514979-d034ce80-647e-11ea-9ba6-23d8e2cfd553.png">

##### Adding a new widget when user is not logged in
<img width="300" src="https://user-images.githubusercontent.com/22608780/76515357-741e7a00-647f-11ea-9ae2-7ff87ff9565b.png">

##### Adding a new widget when v4 stats is not available for a site 
<img width="300" src="https://user-images.githubusercontent.com/22608780/76515684-0888dc80-6480-11ea-94a8-2ced052ed73c.png">


#### Testing
- Run `TodayWidgetListViewModelTest` unit test.
- Add widget to a site that has WC-admin or Woo 4.0 available:
   - Login to a store that has WooCommerce 4.0 or WC-admin installed.
   - Long-press `home screen`.
   - Tap `Widgets` option.
   - Scroll to ***Woo*** widget.
   - Long press the ***Today*** widget and place on the home screen.
   - Notice the current day stats (visitors, order count and revenue) is shown in widget.
   - Click on the widget and notice that you are redirected to the My Store `Today` TAB.
- Verify that the widget data is refreshed when store is changed.
   - Complete the previous test and add a widget to the app.
   - Open the app and switch to a different store.
   - Close the app and notice that the stats data in the widget is changed to reflect the site selected in the previous step.
- Verify that error message is displayed when user is not logged in.
   - Complete the previous test and add a widget to the app.
   - Open the app and logout of the app.
   - Close the app and notice that an error is displayed in the widget when you are logged out.
- Verify that error message is displayed when stats v4 is not available for a site:
   - Complete the first test and add a widget to the app.
   - Open the `wp-admin` of the current site you are logged into.
   - Disable WC-Admin plugin for the site and refresh the app.
   - Notice the stats reverted banner is displayed in the app.
   - Close the app and notice that the stats widget displays an error message.

#### Notes
- Note that is PR is to be merged to a feature branch and will not be available for end users until all the steps in the [master thread](https://github.com/woocommerce/woocommerce-android/issues/2041) is completed.
- This PR only adds the logic to display the stats widget for the current site. The option to support multiple sites will take place in a subsequent PR.
- There are also some TODO's to add tracking events, which will again take place in a separate PR. 

Update release notes:
